marlowe-spec-test/src/Marlowe/Spec/Core/Semantics.hs

```diff
@@ -45,10 +111,238 @@ divisionRoundsTowardsZeroTest interpret = reproducibleProperty "Division roundin
     numerator <- run $ generateT $ genValue interpret
     denominator <- run $ generateT
         (genValue interpret
-          `suchThat` (\d -> (abs_int $ evalValue env state numerator) `less_int` (abs_int $ evalValue env state d))
+          `suchThat` (\d -> (Arith.abs_int $ evalValue env state numerator) `Arith.less_int` (Arith.abs_int $ evalValue env state d))
         )
     let
         req :: Request JSON.Value
         req = EvalValue env state (DivValue numerator denominator)
         successResponse = RequestResponse $ toJSON (0 :: Int)
-    assertResponse interpret req successResponse
\ No newline at end of file
+    assertResponse interpret req successResponse
+
+fixIntervalTest :: InterpretJsonRequest -> TestTree
+fixIntervalTest interpret = reproducibleProperty "Calling fixInterval test" do
+    i@(Arith.Int_of_integer a, Arith.Int_of_integer b) <- run $ generate $ genInterval
+    state <- run $ generateT $ genState interpret
+    let
+        req :: Request JSON.Value
+        req = FixInterval (a, b) state
+        successResponse = RequestResponse $ toJSON $ fixInterval i state
+    assertResponse interpret req successResponse
+
+computeTransactionTest :: InterpretJsonRequest -> TestTree
+computeTransactionTest interpret = reproducibleProperty "Calling computeTransaction test" do
```

Comment:
```suggestion
computeTransactionTest interpret = reproducibleProperty "Compute Transaction" do
```

Reply:
running only this test with tracing enalbed using the following command produced a 13Mb log file.

```
cabal run marlowe-spec -- -c </path/to/cp-client> -p "/Calling computeTransaction test/" --trace-cp t.log
```

I counted 59.500 requests to the client process using this command

```
cat t.log | grep "Send JSON request" | wc
```

Out of which 59.400 where random generation requests. Which makes sense as the remaining 100 requests are the compute transaction.

```
cat t.log | grep "generate-random-value" | wc
```

Out of the 100 requests, 88 returned in error

```
cat t.log | grep "transaction_error" | wc
```

50 TEApplyNoMatchError
2 TEAmbiguousTimeIntervalError
35 intervalInPastError
1 TEUselessTransaction

To be meaningful, this test should only output successful outputs. We should probably also add other tests that forces the different types of errors.